grading/load_config_json.cpp

```diff
@@ -20,6 +21,52 @@ void json_set_default(nlohmann::json &whole_config, std::string field, const T&
   }
 }
 
+uint64_t parse_file_size(std::string size_str) {
+  // Trim whitespaces, it cannot handle UTF-8 characters
+  size_str.erase(std::remove_if(size_str.begin(), size_str.end(), isspace), size_str.end());
+
+  // Convert string to lowercase, it cannot handle UTF-8 characters
+  for (char& c : size_str)
+    c = std::tolower(c);
+
+  // Accepted units, all Bs could be omitted:
+  //    Base 1000: B, KB,  MB,  GB,  TB   (SI)
+  //    Base 1024: B, KiB, MiB, GiB, TiB  (IEC-i)
+  const std::regex accepted { "\\d+([kmgt]i?)?b?" };
+
+  // Extract numeric file size, will cause exception if the number is larger than 64-bit
+  size_t unit_begin  = 0;
+  uint64_t file_size = std::stoull(size_str, &unit_begin);
+  // check if it is 0
+  if (file_size == 0) {
+    std::cout << "Got parsed value " << file_size << " from string: " << size_str << std::endl;
+    throw std::invalid_argument("Wrong file size");
+  }
+  // check if it is well formatted
+  if (!std::regex_match(size_str, accepted)) {
```

Comment:
Can we have this check just after conversion to lower_case, so that we dont get an implicit error from `stdoull` when the file_size starts with a string. (eg.`k123`)